pkg/executor/fscache/functionServiceCache.go

```diff
@@ -170,6 +179,73 @@ func (fsc *FunctionServiceCache) service() {
 	}
 }
 
+// DumpFnSvcCache => dump function service cache data to /tmp directory of executor pod.
+func (fsc *FunctionServiceCache) DumpFnSvcCache(ctx context.Context) error {
+	fsc.logger.Info("dumping function service group cache")
+	fnSvcGroup := fsc.connFunctionCache.GetFnSvcGroup(ctx)
+
+	if len(fnSvcGroup) == 0 {
+		return ferror.MakeError(ferror.ErrorNotFound, "function service not found")
+	}
+	info := []string{}
+	for fn, fnSvcGroup := range fnSvcGroup {
+		for addr, fnSvc := range fnSvcGroup.svcs {
+			info = append(info, fmt.Sprintf("function_name:%s\tfn_svc_address:%s\tactive_req:%d\tcurrent_cpu_usage:%v\tcpu_limit:%v",
+				fn, addr, fnSvc.activeRequests, fnSvc.currentCPUUsage, fnSvc.cpuLimit))
+		}
+	}
+
+	if err := dumpData(info, fsc.logger); err != nil {
+		fsc.logger.Error("error while dumping function service group cache", zap.Error(err))
+		return err
+	}
+
+	fsc.logger.Info("dumped function service group cache")
+	return nil
+}
+
+// check whether /tmp dir exists or not
+// if not exists then create /tmp dir and then dump data to a file, if exists then dump data directly to a file
+// new file will be created on every request with unique id
+func dumpData(data []string, logger *zap.Logger) error {
```

Comment:
Please keep this method in common folder which is accessible to all executor types.
Keep it generic where we can just send content to the method and it would write.

All files should be in folder "/tmp/fission-executor-dump/*.txt"

Reply:
done